src/server/controllers.js

```diff
@@ -0,0 +1,186 @@
+	'ReqApprovingReviews',
+	'ReqCodeOwnerReviews',
+	'ReqConversationResolution',
+	'isPrivate',
+];
+
+const generateQuery = (endCursor, {
+	f,
+}) => {
+	let showForks = false;
+	let showSources = true;
+	let showPrivate = false;
+	let showPublic = true;
+	if (f && f.length) {
+		showForks = f.includes('forks');
+		showSources = f.includes('sources');
+		showPrivate = f.includes('private');
+		showPublic = f.includes('public');
+	}
+	return (
+		`query {
+  viewer {
+	repositories(
+	  first: 100
+	  affiliations: [OWNER, ORGANIZATION_MEMBER, COLLABORATOR]
+	  ${endCursor ? `after: "${endCursor}"` : ''}
+	  ${showForks === showSources ? '' : showForks ? 'isFork: true' : 'isFork: false'}
+	  ${showPrivate === showPublic ? '' : showPublic ? 'privacy: PUBLIC' : 'privacy: PRIVATE'}
+	) {
+	  totalCount
+	  pageInfo {
+		endCursor
+		hasNextPage
+	}
+	nodes {
+		name
+		nameWithOwner
+		defaultBranchRef {
+			name
+			branchProtectionRule {
+				allowsForcePushes
+				allowsDeletions
+				dismissesStaleReviews
+				requiredApprovingReviewCount
+				requiresApprovingReviews
+				requiresCodeOwnerReviews
+				requiresConversationResolution
+				restrictsPushes
+			}
+		}
+		deleteBranchOnMerge
+		hasIssuesEnabled
+		hasProjectsEnabled
+		hasWikiEnabled
+		forkingAllowed
+		isArchived
+		autoMergeAllowed
+		isBlankIssuesEnabled
+		isFork
+		isPrivate
+		isSecurityPolicyEnabled
+		isTemplate
+		licenseInfo {
+			name
+		}
+		mergeCommitAllowed
+		owner {
+			login
+		}
+		rebaseMergeAllowed
+		squashMergeAllowed
+		createdAt
+		updatedAt
+		pushedAt
+		viewerHasStarred
+		viewerPermission
+		viewerSubscription
+	  }
+	}
+  }
+  rateLimit {
+	cost
+	remaining
+  }
+}
+`);
+};
+
+/*
+ * DOUBT: Copy-Pasted this function here as I was having issues with importing the function
+ * from src/commands/detail.js
+ */
+const detail = async (flags) => {
+	if (flags.m) {
+		return listMetrics(getMetrics(metricNames));
+	}
+	let metrics;
+	if (flags.p?.length > 0) {
+		metrics = getMetrics([
+			'Repository',
+			'isFork',
+			'isPrivate',
+			...metricNames.filter((name) => flags.p.includes(name)),
+		]);
+	} else {
+		metrics = getMetrics(metricNames);
+	}
+
+	// Additional Filter on repos
+	let filter;
+	if (flags.f?.length === 1 && flags.f[0] === 'templates') {
+		filter = (repo) => repo.isTemplate;
+	}
+	// Get all repositories
+	const { repositories } = await getRepositories(generateQuery, flags, filter);
+	if (!flags.s) {
+		repositories.sort((a, b) => new Date(b.updatedAt) - new Date(a.updatedAt));
+	}
+	// Generate output table
+	const table = generateDetailTable(metrics, repositories, {
+		actual: flags.actual, all: flags.all, goodness: flags.goodness, sort: flags.s, unactionable: flags.unactionable,
+	});
+
+	if (table) {
+		return table;
+	}
+	return null;
+};
+
+const executeCommand = async (req, res) => {
+	const { command } = req.body;
+	const argv = parse(command);
+	argv.goodness = true;
+	argv.token = GH_TOKEN || GITHUB_TOKEN;
```

Comment:
what if the user passed the token via `--token`?

Reply:
Since this particular chunk of code runs on every request from the frontend, the user will need to pass their token every single time which is a bad UX.

One thing that can be done is to ask the user to pass the token the first time and then we can save it in either `sessionStorage`, `localStorage`, or `cookies`. But of course, that would be a bad idea since the token would persist for any other web pages hosted by the user on localhost:3000.

I think this is the right approach to take the token, from the user's env on every request, instead of saving the token at unsafe places.

Reply:
This is on the server tho - so itd be stored in memory just like an env var, since the server is long-running. In no case should the frontend possess the token.

Reply:
So I guess that resolves this conversation, right? 
We will keep it in the server itself as an env-var and it will be fetched programmatically via line 176

Reply:
Sorry, I must not be being clear.

All of these currently work in the same way:
 - `GH_TOKEN=foo repo-report`
 - `GITHUB_TOKEN=foo repo-report`
 - `repo-report --token=foo`

Thus, all of these must work in the same way:
 - `GH_TOKEN=foo repo-report --gui`
 - `GITHUB_TOKEN=foo repo-report --gui`
 - `repo-report --gui --token=foo`

Reply:
Oh, I see... I will remove the token from the code and pass it this way.